azuredevops/client.go

```diff
@@ -56,26 +56,26 @@ func NewClient(connection *Connection, baseUrl string) *Client {
 		client.Timeout = *connection.Timeout
 	}
 	return &Client{
-		baseUrl:                 baseUrl,
-		client:                  client,
-		authorization:           connection.AuthorizationString,
-		suppressFedAuthRedirect: connection.SuppressFedAuthRedirect,
-		forceMsaPassThrough:     connection.ForceMsaPassThrough,
-		userAgent:               connection.UserAgent,
+		BaseUrl:                 baseUrl,
```

Comment:
Can you share how you plan to use each of these properties?  Once they are set on the connection they are not supposed to be changed.  Clients are cached on the connection, so the properties need to match.  Resource locations are cached on the client, so changing the base url would invalidate it.

Reply:
```
client := &http.Client{Transport: tr}
if connection.Timeout != nil {
	client.Timeout = *connection.Timeout
}

c :=  &azuredevops.Client{
	baseUrl:                 connection.BaseUrl,
	client:                  client,
	authorization:           connection.AuthorizationString,
	suppressFedAuthRedirect: connection.SuppressFedAuthRedirect,
	forceMsaPassThrough:     connection.ForceMsaPassThrough,
	userAgent:               connection.UserAgent,
}
```

currently this is not feasible. as these are not exposed variables.

Reply:
I agree to the point changing properties is not the way intended. Creating Client with own configurable options should be exposed.

Reply:
The intention is to create a new connection if you want to change these.  Most apps should never need more than one connection unless you are trying to connect to multiple organizations or trying to sign in with multiple identities.

Reply:
Agree. Current issue is, http client is internally created, this limits the customisation of standard http client creation (specially TLS Client).

As mentioned two approaches, we can add NewTLSClient(c Connection, baseUrl string, tls tls.Config)
or the above approach. I am kinda favouring the NewTLSClient the more I think about it.

Reply:
Can we create a new constructor for the Connection to take a delegate that will be used to construct the client?  This way we ensure a consistent http client is used for each http client created from the Connection.

Reply:
Can you please review the changes now?